src/hal/src/command/mod.rs

```diff
@@ -604,6 +604,46 @@ pub trait CommandBuffer<B: Backend>: fmt::Debug + Any + Send + Sync {
     /// Requests a timestamp to be written.
     unsafe fn write_timestamp(&mut self, stage: pso::PipelineStage, query: query::Query<B>);
 
+    /// TODO docs
+    /// `build_range_infos` must be the same length as `infos` and each element must have a length equal to the parallel `info` entry's `geometries.len()`. there's probably a way to make this safer without mangling the api shape too much...
+    unsafe fn build_acceleration_structures(
+        &self,
+        infos: &[acceleration_structure::AccelerationStructureGeometryDesc<B>],
+        build_range_infos: &[&[acceleration_structure::AccelerationStructureBuildRangeDesc]],
+    );
+
+    /// TODO docs
+    /// `indirect_device_addresses` is an array of infoCount buffer device addresses which point to infos[i]->geometryCount VkAccelerationStructureBuildRangeInfoKHR structures defining dynamic offsets to the addresses where geometry data is stored, as defined by infos[i].
```

Comment:
similarly, can we merge all the things that are of the same length into a structure, so that it's an iterator of this structure that the user provides?